### PR TITLE
discount: 2.2.7 -> 2.2.7b

### PR DESCRIPTION
**Pull request by McSinyx**
Also install man pages and sample programs

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/discount/default.nix [hunk 1]
```diff
@@ -1,36 +1,33 @@
...
+    "--pkg-config"
+    "--h1-title"
   ];
 
   enableParallelBuilding = true;
+
+  installPhase = ''
+    runHook preInstall
+    make install.everything
+    runHook postInstall
```
Comment (trofi) [inline pkgs/tools/text/discount/default.nix:28]: You can also try slightly shorter `installTargets = [ "install.everything" ]`

**Review by trofi — APPROVED** [review]

File: pkgs/tools/text/discount/default.nix (hunk 1 above)
Comment (McSinyx) [reply pkgs/tools/text/discount/default.nix]: Thanks, TIL!

**Comment by trofi** [comment]
Result of `nixpkgs-review pr 173345` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt512.discover</li>
    <li>libsForQt512.kdepim-addons</li>
    <li>libsForQt512.kmail</li>
    <li>libsForQt512.okular</li>
    <li>libsForQt514.discover</li>
    <li>libsForQt514.kdepim-addons</li>
    <li>libsForQt514.kmail</li>
    <li>libsForQt514.okular</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>calligra</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>discount</li>
    <li>ejson</li>
    <li>haskellPackages.discount</li>
    <li>jsduck</li>
    <li>kile</li>
    <li>libsForQt5.discover (libsForQt515.discover ,plasma5Packages.discover)</li>
    <li>libsForQt5.kdepim-addons (libsForQt515.kdepim-addons ,plasma5Packages.kdepim-addons)</li>
    <li>libsForQt5.kmail (libsForQt515.kmail ,plasma5Packages.kmail)</li>
    <li>libsForQt5.okular (libsForQt515.okular ,plasma5Packages.okular)</li>
    <li>minder</li>
    <li>notes-up</li>
    <li>pdfpc</li>
    <li>thiefmd</li>
  </ul>
</details>

**Comment by trofi** [comment]
`calligra` failure looks unrelated and is probably an existing failure:

```
/build/calligra-3.2.1/filters/karbon/pdf/SvgOutputDev.cpp:413:36: error: cannot convert 'const std::shared_ptr<GfxFont>' to 'GfxFont*' in initialization
  413 |     GfxFont * font = state->getFont();
      |                      ~~~~~~~~~~~~~~^~
      |                                    |
      |                                    const std::shared_ptr<GfxFont>
```

**Comment by trofi** [comment]
Pulled as is. Thank you!

**Pull request merged**